### PR TITLE
replay: troubleshooting performance

### PR DESCRIPTION
**Pull request by bruno-garcia**
Resolves: https://github.com/getsentry/sentry-docs/issues/6887

### CODE REVIEW

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 1]
```diff
@@ -44,6 +44,22 @@ Another reason you might not see the body is if the format is not supported. We
...
+The Sentry SDK attempts to minimize potential [performance overhead in a few different ways](/product/session-replay/performance-overhead/#how-is-session-replay-optimized).
+One such way is by keeping track of how many DOM mutations are happening. This is important because many mutations can slow down the web page. And for Replay to keep track of it,
+it could inccur an additional slow down. This is an indication your page could be optimized. For example, a drop down list with thousands of entries could be refactored into one with lazy loading.
+Fetching more data and updating the drop down list as the user scroll through it.
+The SDK has a <PlatformLink to="/session-replay/configuration/#mutation-limits">configuration that allows you configure the limits</PlatformLink> before recordings stops.
+
+## The SDK is slowing down my website
+
+Sentry's SDK latest version includes several performance improvements. Two of the major culprits for performance impact we've identified
+were due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these
```
Comment (bruno-garcia) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:58]:
```suggestion
where due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these
```

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 2]
```diff
@@ -44,6 +44,22 @@ Another reason you might not see the body is if the format is not supported. We
...
+it could inccur an additional slow down. This is an indication your page could be optimized. For example, a drop down list with thousands of entries could be refactored into one with lazy loading.
+Fetching more data and updating the drop down list as the user scroll through it.
+The SDK has a <PlatformLink to="/session-replay/configuration/#mutation-limits">configuration that allows you configure the limits</PlatformLink> before recordings stops.
+
+## The SDK is slowing down my website
+
+Sentry's SDK latest version includes several performance improvements. Two of the major culprits for performance impact we've identified
+where due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these
+cases such as [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recording on pages that have excessively high number of mutations.
+It's important to update the SDK to the latest version to benefit from these improvements. In the event you encounter performance problems with Session Replay,
```
Comment (bruno-garcia) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:60]:
```suggestion
It's important to update the SDK to the latest version to benefit from these improvements. In the event you encounter performance problems with Session Replay on the latest version of the SDK,
```

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 3]
```diff
@@ -44,6 +44,22 @@ Another reason you might not see the body is if the format is not supported. We
...
+The SDK has a <PlatformLink to="/session-replay/configuration/#mutation-limits">configuration that allows you configure the limits</PlatformLink> before recordings stops.
+
+## The SDK is slowing down my website
+
+Sentry's SDK latest version includes several performance improvements. Two of the major culprits for performance impact we've identified
+where due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these
+cases such as [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recording on pages that have excessively high number of mutations.
+It's important to update the SDK to the latest version to benefit from these improvements. In the event you encounter performance problems with Session Replay on the latest version of the SDK,
+please [open a GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) describing the situation so we can further investigate.
+
```
Comment (ryan953) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:62]:
```suggestion
## The SDK is slowing down my website

Sentry's ReplaySDK v??? and above includes several performance improvements. Two of the major culprits for performance impact we've identified are due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these cases such as [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recording on pages that have excessively high number of mutations. It's important to update the SDK to the latest version to benefit from these improvements. In the event you encounter performance problems with Session Replay on the latest version of the SDK, please [open a GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) describing the situation so we can further investigate.

```

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 4]
```diff
@@ -44,6 +44,22 @@ Another reason you might not see the body is if the format is not supported. We
 
 More details about this feature can be found in the <PlatformLink to="/session-replay/configuration/">configuration page</PlatformLink>.
 
+## A large number of mutations was detected ( N ). This can slow down the Replay SDK and impact your customers.
```
[the remark this answers is not in the document]
Comment (bruno-garcia) [reply src/platforms/javascript/common/session-replay/troubleshooting.mdx]: Makes sense, added

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx (hunk 3 above)
Comment (bruno-garcia) [reply src/platforms/javascript/common/session-replay/troubleshooting.mdx]: I edited ur suggestion to remove the reference to the exact version (XXX) in favor of leaving it open again. I understand this isn't exact, but I'm purposely trying to just say "the latest" to hint nudge folks into updating and verifying whether it's slow still, that's because we continue to add more perf improvements constantly

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 5]
```diff
@@ -44,6 +44,16 @@ Another reason you might not see the body is if the format is not supported. We
...
+
+## The SDK is slowing down my website
+
+Sentry's latest SDK version include several performance degradation safeguards. Two of the major culprits for performance impact we've identified are due to websites having extremely large number of mutations happening (see above _large number of mutations_), and very large console messages. We've added improvements for these cases such as [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recording on pages that have excessively high number of mutations. It's important to update the SDK to the latest version to benefit from these improvements. In the event you encounter performance problems with Session Replay on the latest version of the SDK, please [open a GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) describing the situation so we can further investigate.
```
Comment (lizokm) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:55]:
```suggestion
We've identified two major potential reasons for impacted performance and have added several performance degradation safeguards as a result, starting with version x.xx of the Sentry SDK. The two culprits are websites that have a large numbers of mutations (Read the entry above to learn more.) and large console messages. 

We've added improvements, including [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recordings on pages that have a large number of mutations. To see these updates in action, be sure you have the latest version of our SDK.

If you're having any problems with the latest SDK version, we want to hear about it. Please open a [GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) and describe your situation. 
```

**Review by lizokm — APPROVED** [review]
Added some language suggestions

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (bruno-garcia) [reply src/platforms/javascript/common/session-replay/troubleshooting.mdx:55]: It'll never be out of date, because it's the 'latest'. I'm trying to indicate to folks to update to the latest SDK whatever it'll be. Otherwise we'll need to update the version on each SDK improvement and we're doing these weekly.

I could reword it to indicate that if it's not clear.

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 6]
```diff
@@ -44,6 +44,20 @@ Another reason you might not see the body is if the format is not supported. We
...
+
+If you're seeing the "A large number of mutations was detected" message, it means that your page could be optimized. For example, a dropdown list with thousands of entries could be refactored into one with lazy-loading. This would fetch additional data and update the dropdown list as the user scrolls through it. The SDK has a <PlatformLink to="/session-replay/configuration/#mutation-limits">configuration</PlatformLink> that allows you to configure the limits before recording stops.
+
+## The SDK is slowing down my website.
+
+We've identified two major potential reasons for impacted performance and have added several performance degradation safeguards as a result, starting with version x.xx of the Sentry SDK. The two culprits are websites that have a large numbers of mutations (Read the entry above to learn more.) and large console messages.
```
Comment (bruno-garcia) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:55]:
```suggestion
We've identified two major potential reasons for impacted performance and have added several performance degradation safeguards as a result. Please update the SDK to the latest version to benefit from all these improvements. The two culprits are websites that have a large numbers of mutations (Read the entry above to learn more.) and large console messages. 
```

**Comment by bruno-garcia** [comment]
@lizokm I requested a review again because your last review asked for an SDK version specified and I took a different route. I'm trying to not get back to these docs every time we have a bugfix or an additional refactor to improve performance, and instead indicate that folks need to update to the latest version to benefit from performance improvements. 

We receive far too many bug/performance reports from folks that didn't bother updating the SDK before reporting, only to find out that their issue has been resolved. That's the motivation to leave this open indicating an update to the latest version.

File: src/platforms/javascript/common/session-replay/troubleshooting.mdx [hunk 7]
```diff
@@ -44,6 +44,20 @@ Another reason you might not see the body is if the format is not supported. We
...
+
+## The SDK is slowing down my website.
+
+We've identified two major potential reasons for impacted performance and have added several performance degradation safeguards as a result. Please update the SDK to the latest version to benefit from all these improvements. The two culprits are websites that have a large numbers of mutations (Read the entry above to learn more.) and large console messages.
+
+We've added improvements, including [truncating the console](https://github.com/getsentry/sentry-javascript/pull/7917) and disabling Replay recordings on pages that have a large number of mutations. To see these updates in action, be sure you have the latest version of our SDK.
+
+If you're having any problems with the latest SDK version, we want to hear about it. Please open a [GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) and describe your situation.
+
```
Comment (ryan953) [inline src/platforms/javascript/common/session-replay/troubleshooting.mdx:60]: I agree that reminding people they should be on the latest version is very important. Lets put that up at the top.

Also, i re-ordered the two sections, I think the generic 'perf' one can go first, for people reading top-to-bottom, then the specific perf stuff afterwards. For people going a google search for specific terms or error messages it won't matter.

```suggestion
## The SDK is slowing down my website.

If you are experiencing slowdowns on your website first make sure you update to the latest SDK version. This will include the most recent bug fixes and performance improvements.

We've identified two major potential reasons for impacted performance and have added several performance degradation safeguards as a result. The two culprits are websites that have a large numbers of mutations (Read the entry below to learn more.) and large console messages.

To mitigate these problems SDK version 7.54.0 will [truncate console messages](https://github.com/getsentry/sentry-javascript/pull/7917) and will disabled Replay recordings on pages that have a large number of mutations.

If you're having any problems with the latest SDK version, we want to hear about it. Please open a [GitHub issue](https://github.com/getsentry/sentry-javascript/issues/new?assignees=&labels=Type%3A+Bug&projects=&template=bug.yml) and describe your situation.

## I see the message: "A large number of mutations was detected (N). This can slow down the Replay SDK and impact your customers."

The Sentry SDK attempts to minimize potential [performance overhead](/product/session-replay/performance-overhead/#how-is-session-replay-optimized) in a few different ways. For example, by keeping track of how many DOM mutations are happening, then disabling recording when a large number of changes are detected. Many simultaneous mutations can slow down a web page whether Session Replay is installed or not, but when a large number of mutations happen, the Replay client can incur an additional slowdown because it records each change.

If you're seeing the "A large number of mutations was detected" message, it means that your page could be optimized. For example, a dropdown list with thousands of entries could be refactored into one with lazy-loading. This would fetch additional data and update the dropdown list as the user scrolls through it. The SDK has a <PlatformLink to="/session-replay/configuration/#mutation-limits">configuration</PlatformLink> that allows you to configure the limits before recording stops.
```

Comment (lizokm) [reply src/platforms/javascript/common/session-replay/troubleshooting.mdx:60]: I made a few small tweaks here ^^

**Review by lizokm — APPROVED** [review]
Made a few tweaks to Ryan's update. This LGTM :)

**Comment by bruno-garcia** [comment]
Thanks everyone
